### DISCUSSION TITLE
734

### DISCUSSION THREAD

**Discussion by giscus** [jojoldu/blog-comments#914] [Comments]
# 734

좋은 예외 처리는 견고한 프로그램을 만들고, 좋은 사용자 경험을 줄 수 있다. 예외 처리를 통해 애플리케이션이 예기치 않게 종료되는 것을 방지하고, 갑작스런 종료 대신 사용자는 무엇이 잘못되었는지, 그리고 가능하다면 어떻게 바로잡을 수 있는지에 대한 의미 있는 오류 메시지를 받을 수 있다. 뿐만 아니라 좋은 예외처리는 개발자가 문제를 진단하는 데 큰 도움이 되어 이로 인해 문제 해결 시간이 단축된다. 특히, 복잡한 시스템에서 여러 단계의 프로세스가 있는 경우 예외는 프로세스의 위치에 따라 다르게 처리되어 적절한 예외 처리는 이러한 프로그램의 프로세스를 관리하는 데 유연성을 제공한다. 반면, 이를 위해 과도하게 사용하면 메인 비즈니스 로직이 무엇인지 파악하기 힘들 정도로 너무 많은 오류 처리를 가지고 있는..

https://jojoldu.tistory.com/734

**Comment by dangddoong** [comment]
좋은 글에 늘 감사드립니다. 많이 배우고 있습니다. 
-----
글 초반에 `복구 가능한 오류와 복부 불가능한 오류를 구분하는 것이다` <- '복부 불가능'이라는 오타가 있어 댓글남깁니다. 
-----
좋은 하루 되세용 감사합니다 ^^

**Reply by jojoldu** [reply]
앗 감사합니다 ㅎㅎ 수정해두었습니다.

**Comment by moonqqqq** [comment]
오늘도 제 코드에 바로 적용하기 아주 좋을 내용을 알려주시네요 ! 감사합니다 !

**Comment by bnitech** [comment]
현업에서 늘 고민하던 부분인데, 이렇게 정리해주셔서 감사합니다.
많이 배웠습니다. ( _ _ )

**Comment by jongsu253** [comment]
좋은 동료를 둔 팀원들이 부럽네요 👍

내부 사정은 잘 모르지만
아마도 최근에 팀원들이 가장 고민을 많이하던 포인트에 대해 내용을 정리해주신 것 같은데,
고민 해결을 위한 방향성 제시와 내용 정리를 해주신 것 같다는 생각이 드네요!

가슴 따뜻한 개발자의 온정에 감동받고 갑니다 😁

**Comment by awdsza** [comment]
리팩토링하면서 조금씩 적용해보고싶은부분인데 정리해주셔서 감사합니다.

**Comment by jinwook3** [comment]
안녕하세요 향로님 ! 블로그 글을 읽고 많이 배우고 있습니다. 감사합니다!

추신) `예외에 해당 예외의 근본 원인을 **찾알** 수 있는 정확한 정보를 남겨준다.` <- 오타가 있습니다.

**Reply by jojoldu** [reply]
아고 감사합니다! 바로 수정해두겠습니다 :)

**Comment by zieunx** [comment]
항상 감사드립니다!
정리가 너무 깔끔해서 자주 참고하겠습니다 :)

--- 

**<오타 제보>**

> 외부 라이브러리 (pay.billing) 에서 발생하는 문제와 우리가 관리하는 코드는 (databse.save) 가 같은 방식으로 해결해서는 안된다.

이 문장에서 `databse.save` 오타가 있습니다!

**Comment by kyungjunleeme** [comment]
글 잘 봤습니다. 하지만, 제가 이 글을 읽을 때 약간 헷갈리는 부분이, 예외랑 오류를 엄격하게 구분하고 사용하셔서 작성하지 않으신 것 같습니다. 그래서, 이 글에서는 오류라는 말은 제거를 해도 무방할 것 같습니다. 아니면 좀 더 명확하게 구분을 해주시면 좋을것 같습니다. 


> 물론 이 때 원인이 되는 Exception을 상위 Exception의 생성자에 넘기는 exception chaning기법도 사용할 수 있다.

chaning -> chaining (오타 제보 드립니다)
